src/ray/core_worker/core_worker.cc

```diff
@@ -2445,11 +2436,18 @@ void CoreWorker::HandleSubscribeForObjectEviction(
 
     object_status_publisher_->Publish(rpc::ChannelType::WORKER_OBJECT_EVICTION,
                                       pub_message, object_id.Binary());
-    object_status_publisher_->UnregisterSubscription(
-        rpc::ChannelType::WORKER_OBJECT_EVICTION, subscriber_node_id, object_id.Binary());
   };
 
-  ObjectID object_id = ObjectID::FromBinary(request.object_id());
+  const auto object_id = ObjectID::FromBinary(message.object_id());
+  const auto intended_worker_id = WorkerID::FromBinary(message.intended_worker_id());
+  if (intended_worker_id != worker_context_.GetWorkerID()) {
+    RAY_LOG(INFO) << "The SubscribeForObjectEviction message is for "
+                  << intended_worker_id << ", but the current worker id is "
+                  << worker_context_.GetWorkerID() << ". This will be no-op.";
+    unpin_object(object_id);
```

Comment:
We need to publish the message. Previously, it just replies the failure status, but this won't work for the command batch (command batch should always succeed)